tests/php-unit-tests/unitary-tests/setup/ModuleInstallerAPITest.php

```diff
@@ -240,4 +240,48 @@ public function testMoveColumnInDB_MoveMultipleTable(): void
 		$this->assertEquals('from table 1', $sFromTable1Data, "Data was not moved as expected");
 		$this->assertEquals('from table 2', $sFromTable2Data, "Data was not moved as expected");
 	}
+
+	/**
+	 * Test that moving columns from/to the same table works
+	 *
+	 * @covers       \ModuleInstallerAPI::MoveColumnInDB
+	 *
+	 * @return void
+	 * @throws \ArchivedObjectException
+	 * @throws \CoreException
+	 * @throws \MySQLException
+	 * @throws \MySQLHasGoneAwayException
+	 * @throws \MySQLQueryHasNoResultException
+	 */
+	public function testMoveColumnInDB_SameTable(): void
+	{
+		// Info from the original table
+		$sOrigClass = "Person";
+		$sOrigAttCode = "first_name";
+		[$sOrigTable, $sOrigColName] = $this->GetInfoFromTable($sOrigClass, $sOrigAttCode);
+
+		// Info for the destination column
+		$sDstNonExistingColName = "non_existing_column";
+
+		// Save value from original table as a reference
+		$oPerson = MetaModel::GetObject($sOrigClass, 1);
+		$sOrigValue = $oPerson->Get($sOrigAttCode);
+
+		// Try to move data
+		ModuleInstallerAPI::MoveColumnInDB($sOrigTable, $sOrigColName, $sOrigTable, $sDstNonExistingColName);
+
+		// Check if data was actually moved
+		// - Either way, the column should exist
+		$sDstValue = CMDBSource::QueryToScalar(
+			<<<SQL
+SELECT `{$sDstNonExistingColName}` FROM `{$sOrigTable}` WHERE `id` = 1
+LIMIT 1
+SQL
+		);
+
+		// Put data back in the original table
+		ModuleInstallerAPI::MoveColumnInDB($sOrigTable, $sDstNonExistingColName, $sOrigTable, $sOrigColName);
+
+		$this->assertEquals($sOrigValue, $sDstValue, "Data was not moved as expected");
```

Comment:
```suggestion
		// Try to move data
		ModuleInstallerAPI::MoveColumnInDB($sOrigTable, $sOrigColName, $sOrigTable, $sDstNonExistingColName);

		// Check if data was actually moved
		// - Either way, the column should exist
		$sDstValue = CMDBSource::QueryToScalar(
			<<<SQL
SELECT `{$sDstNonExistingColName}` FROM `{$sOrigTable}` WHERE `id` = 1
LIMIT 1
SQL
		);

		$this->assertEquals($sOrigValue, $sDstValue, "Data was not moved as expected");

		// Put data back in the original table
		ModuleInstallerAPI::MoveColumnInDB($sOrigTable, $sDstNonExistingColName, $sOrigTable, $sOrigColName);
```

I would rather put the assert there with your actual check. With the current proposition, it confused me, as I though you checked the data was put back in it's original column.

Reply:
Otherwise the test is good, thanks!

Reply:
Doen't it break the state of the table when assert fails? (The way unit tests in php work are weird and confusing anyways)

Reply:
I wanted to keep consistency with the [code written by my ex Lead dev](https://github.com/Combodo/iTop/pull/676/files/007927c852879001fd30552b2ce6322b8371d911#diff-26ac6d70dfb1aa45415774deb778df56af8238e996d5d39f46cdc79cdef36e2dL137-L144)
Joking aside, wouldn't a failed assert stop the test execution and leave an inconsistent table?

Reply:
🙈🙈🙈

I think both your statements are valid, I therefore redraw mine 😁